### PR TITLE
stable/velero: Update Velero to v1.2.0

### PR DESCRIPTION
**Pull request by carlisia**
Signed-off-by: Carlisia <carlisia@vmware.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/velero/values.yaml
Comment (prydonius) [inline stable/velero/values.yaml]: Can we hold this change back for now? There's this PR that is on hold that we can use to update to 1.2.0 and add support for cross-subscription backups in azure https://github.com/helm/charts/pull/17585/files

**Review by prydonius — APPROVED** [review]
This is great, thanks @carlisia. If we can remove the Velero version change, we can just do a patch bump on the chart here

**Comment by prydonius** [comment]
Ah the plugin docs is relevant to 1.2.0, so we should hold this and use this PR to release 1.2.0. https://github.com/helm/charts/pull/17585 can then rebase once this is merged.

/hold

File: stable/velero/README.md
Comment (skriss) [inline stable/velero/README.md]: bump to `v1.0.0`

File: stable/velero/README.md
Comment (skriss) [inline stable/velero/README.md]: Can remove this line, as it's not relevant beyond v1.0

**Review by skriss — CHANGES_REQUESTED** [review]
Looks great! couple minor things

File: stable/velero/README.md
Comment (nrb) [inline stable/velero/README.md:42]: Is this information added elsewhere? There's the CLI command example, but it lacks the description/required fields from this table, which I find useful.

Comment (carlisia) [reply stable/velero/README.md:42]: We had talked about this in the past in the context of the `velero install` command. What we had decided was to have the CLI itself be the source of documentation, instead of trying to keep such documentation up to date in multiple places. At least this is my recollection, and also doesn't mean we can't change it.

Comment (nrb) [reply stable/velero/README.md:42]: Yeah, that's true of `velero install`, though the Helm chart is a bit different. For example, all of these options can (an normally, should) be in a values.yaml file versus just being on the command line. Also, when they're on the command line, they're not their own flags, but rather key/value pairs given to `set`.

There's also options like `credentials.existingSecret` that aren't in `velero install` that have some interactions with other values. That's not immediately clear from just the command line instructions here.

Finally, there's also [Helm standards](https://helm.sh/docs/chart_best_practices/#document-values-yaml) that mention documenting what's possible in a `values.yaml` file.

I think I'd propose having both the CLI example that was added as well as the table here.

Comment (carlisia) [reply stable/velero/README.md:42]: Gotcha. But these values are all documented in `values.yaml`. Do they need to be duplicated in the readme too?

**Comment by billimek** [comment]
Should we include a facility within velero to allow for the helm chart to install the necessary plugin(s) which now require the `--plugins` argument to be passed to the velero container?

See [v1.2.0 release notes](https://github.com/vmware-tanzu/velero/releases/tag/v1.2.0)

**Comment by prydonius** [comment]
@billimek the docs have been updated to reflect that in this PR: https://github.com/helm/charts/pull/18492/files#diff-600a674d7d9abb8a92992fae1f8bc6d5R54. I do think we can make this UX a little nicer by making it simpler to specify plugin images, but maybe something we can address in a follow up PR?

**Comment by thabach** [comment]
I commented on Google groups in regards to the (old, the 1.1.0) Helm chart yesterday, see here https://groups.google.com/forum/#!topic/projectvelero/pUm3CyrIusA 

A fix for the first issue should be straightforward and I would love to supply a pull but where-to ? The second issue I am not sure if you guys feel like that should be steered by the Helm chart, as customisation can done via a CRD according to https://velero.io/docs/master/restic/ , it would certainly make things simpler.

> Heya
> 
> great work and congrats to the new 1.2 release !
> 
> I happen to run into two issues with the Helm chart https://github.com/helm/charts/tree/master/stable/velero for velero v1.1.0 (the latest available).
> the restic.podVolumePath configuration is not taken into account at all, but the default gets always applied
> while the image.repository setting is taken into account in assembling the velero image path, but not in addressing the velero-restic-restore-helper init-container (which prevents me from doing a restore in the first place within our corporate image caching setup)
> Maybe that could be fixed for the soon to be released v1.2.0 helm chart, I would certainly highly appreciate.
> 
> Kind regards, Christian.

File: stable/velero/README.md
Comment (nrb) [reply stable/velero/README.md:42]: No, I don't think they need to duplicated - in the `values.yaml` is fine. I'll mark this as resolved.

**Comment by nrb** [comment]
@thabach Thanks for reporting those problems. Could you file those as separate issues in this repo and assign them to me, so we can get them addressed? I don't know that we'll include them in this PR, but we may do a follow-up point release.

**Comment by thabach** [comment]
> @thabach Thanks for reporting those problems. Could you file those as separate issues in this repo and assign them to me, so we can get them addressed? I don't know that we'll include them in this PR, but we may do a follow-up point release.

@nrb just realised that I was sitting on an old `restic-daemonset.yaml`-template and the first bullet was already resolved some time ago - sorry for the false alarm. For the second issue I will report a separate issue.

**Comment by prydonius** [comment]
@carlisia are you able to rebase this and fix conflicts?

**Comment by carlisia** [comment]
Getting this rebased now..

**Comment by carlisia** [comment]
@prydonius / @nrb PTAL.

File: stable/velero/README.md
Comment (skriss) [inline stable/velero/README.md:76]: Let's add an "Upgrading to v1.2.0" subsection here.  I think we can also remove the "Upgrading to v0.11.0" section, and I'd be fine dropping the "Upgrading to v1.0.0" section as well.

**Comment by skriss** [comment]
otherwise, LGTM!

**Comment by prydonius** [comment]
@carlisia looks like this is good to go once the conflicts are resolved!

File: stable/velero/Chart.yaml
Comment (JoaoBraveCoding) [inline stable/velero/Chart.yaml]:
```suggestion
version: 2.7.0
```

File: stable/velero/values.yaml
Comment (JoaoBraveCoding) [inline stable/velero/values.yaml]:
```suggestion
  #       name: plugins
```

File: stable/velero/values.yaml
Comment (JoaoBraveCoding) [inline stable/velero/values.yaml]:
```suggestion
  #   image: velero/velero-plugin-for-aws:v1.0.0
```

Comment (ZeroVEVO) [reply stable/velero/values.yaml]: where do I need to specify the plugins inside of the helm chart. Is there a reference to the values file? Actualy im having an issue upgrading.
https://github.com/helm/charts/issues/19427

Comment (JoaoBraveCoding) [reply stable/velero/values.yaml]: You have to specify it in the `initContainers` field in the values file like is exemplified [here](https://github.com/carlisia/charts/blob/c-velero-1.2.0-beta/stable/velero/values.yaml#L23). Hope this was the answer you were looking for.

File: stable/velero/values.yaml
Comment (ZeroVEVO) [inline stable/velero/values.yaml:22]: this would be very helpful, otherwise i wont know where to add the plugins to the chart

**Review by ZeroVEVO — APPROVED** [review]

File: stable/velero/values.yaml
Comment (ZeroVEVO) [reply stable/velero/values.yaml]: I could solve the problem by using init containers:
```
initContainers:
  - name: velero-plugin-for-microsoft-azure
    image: velero/velero-plugin-for-microsoft-azure:v1.0.0
    volumeMounts:
      - name: plugins
        mountPath: /target
```

**Comment by skriss** [comment]
@carlisia let's try to get this rebased and merged soon so we get updated and don't continue to hit merge conflicts!

File: stable/velero/README.md
Comment (invidian) [reply stable/velero/README.md:42]: I think the convention is to actually have them documented. Personally I find dot notation way better searchable than YAML notation. Example:

`configuration.backupStorageLocation.config.storageAccount` 
vs
```
configuration:
  # a lot of documentation
  # a lot of documentation
  # a lot of documentation
  # a lot of documentation
  # a lot of documentation
  backupStorageLocation:
    # a lot of documentation
    # a lot of documentation
    # a lot of documentation
    # a lot of documentation
    # a lot of documentation
    config:
      # a lot of documentation
      # a lot of documentation
      # a lot of documentation
      # a lot of documentation
      # a lot of documentation
      storageAccount

Comment (nrb) [reply stable/velero/README.md:42]: That's a good point. While it's well taken, I think in order to get Helm users onto v1.2, we should defer this discussion to a separate PR and focus this one on any changes necessary to get 1.2 installable at this stage.

File: stable/velero/values.yaml
Comment (ZeroVEVO) [reply stable/velero/values.yaml]: @JoaoBraveCoding yes, thats it. I think I didn´t see your message on Friday, but today. Thank you!

**Pull request merged**
